File: es5/bubbleSort.js
//This is the bubblesort algorithm

function bubbleSort(x){
    var temp;
    var temp2;
    var swap=false;
    var length=x.length;
    var count=0;
    for(var i=0;i<length;i++){
        temp=x[i];
        temp2=x[i+1];
        if(x[i]>x[i+1]){
            x[i]=temp2;
            x[i+1]=temp;
            swap=true;
        }
        while(i===length-1 && swap===true){
            i=-1;
            swap=false;
            length--;
        }
    }
    return x;
}

var arr=[100,2,71,72,60,45];
var array=[6,5,4,3,2,1];

console.log(bubbleSort(arr));
console.log(bubbleSort(array));

File: C/bubble_sort.c
#include <stdio.h>

int main(void)
{
    int arr[6] = {78,2,4,1,7,6};
	int temp;
	int temp2;
	int check = 0;
	int arr_len = sizeof(arr) / sizeof(arr[0]);

	while(check < (arr_len - 1))
	{
		check = 0;
		for(int i = 0, n = (arr_len -1); i < n;i++)
		{
			if(arr[i] > arr[i+1])
			{
				temp = arr[i];
				temp2 = arr[i+1];
				arr[i] = temp2;
				arr[i+1] = temp;
			}
			else{
				check++;
			}
		}
	}
    
    for(int j = 0; j < arr_len;j++)
    {
        printf("%i ",arr[j]);
    }
    printf("\n");
    return 0;
}
